### DISCUSSION TITLE
在淘宝小程序中，无法获取材质

### DISCUSSION THREAD

**Discussion by flyingflyfly** [galacean/discussion#14] [Mini Program（小程序）] [answered]
1、模型：https://next-ubanner.oss-cn-beijing.aliyuncs.com/AE_BODY_MOVIN/TMP/2afcc40b678752f3c547a13124f7bef2/0002.gltf
2、`const renderer = entity.getComponent(MeshRenderer);  // renderer为undefined`

**Comment by singlecoder** [comment, marked as answer]
你这个模型完整获取材质方式如下：

```typescript
const gltf = await engine.resourceManager.load<GLTFResource>(
    "https://gw.alipayobjects.com/os/OasisHub/267000040/9994/%25E5%25BD%2592%25E6%25A1%25A3.gltf"
  );
const gltfEntity = gltf.defaultSceneRoot;
rootEntity.addChild(gltfEntity);
// 获取有 MeshRenderer 组件的子节点
const meshEntity = gltfEntity.children[0];
const meshRenderer = meshEntity.getComponent(MeshRenderer);
const material = meshRenderer.getMaterial();
```

原因：

模型加载后，美术给出模型并不一定就是挂在根节点，这个时候最好可以打印下查看子节点，打印 gltf.defaultSceneRoot 可以查看加载模型的结构，如下：

![image](https://user-images.githubusercontent.com/7200304/187888980-32b14b78-f1f7-4133-89cd-caa2c3720ff5.png)

也可以通过 API 直接查找到所有 MeshRenderer，如下：

```typescript
const result: Array<MeshRenderer> = [];
gltf.defaultSceneRoot.getComponentsIncludeChildren(MeshRenderer, result);
```
